src/main/java/com/coveros/selenified/services/Check.java

```diff
@@ -270,6 +321,95 @@ boolean checkContains(JsonElement expectedJson) {
         return pass;
     }
 
+    /**
+     * Checks the actual response json payload contains a key containing a JsonObject
+     * containing each of the pair values provided. The jsonKeys should be passed in
+     * as crumbs of the keys leading to the field with
+     * the expected value. This result will be written out to the output file.
+     *
+     * @param jsonKeys      - the crumbs of json object keys leading to the field with the expected value
+     * @param expectedPairs a hashmap with string key value pairs expected in the json
+     *                      response
+     */
+    @SuppressWarnings("squid:S1201")
+    abstract void contains(List<String> jsonKeys, Map<String, Object> expectedPairs);
+
+    /**
+     * Checks the actual response json payload contains a key containing a JsonObject
+     * containing each of the pair values provided. The jsonKeys should be passed in
+     * as crumbs of the keys leading to the field with
+     * the expected value. This result will be written out to the output file.
+     *
+     * @param jsonCrumbs    - the crumbs of json object keys leading to the field with the expected value
+     * @param expectedPairs a hashmap with string key value pairs expected in the json
+     *                      response
+     */
+    boolean checkContains(List<String> jsonCrumbs, Map<String, Object> expectedPairs) {
+        JsonObject actualValue = this.response.getObjectData();
+        for (String jsonCrumb : jsonCrumbs) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS135)